src/api/route-services/pipelines.js

```diff
@@ -0,0 +1,196 @@
+const AWS = require('aws-sdk');
+const crypto = require('crypto');
+const fetch = require('node-fetch');
+const YAML = require('yaml');
+const jq = require('jq-web');
+const config = require('../../config');
+
+
+class PipelinesService {
+  constructor() {
+    this.stepFunctions = new AWS.StepFunctions({
+      region: config.awsRegion,
+    });
+
+    this.clusterInfo = null;
+  }
+
+  async getClusterInformation() {
+    if (!this.clusterInfo) {
+      const eks = new AWS.EKS({
+        region: config.awsRegion,
+      });
+
+      const accountId = await config.awsAccountIdPromise();
+      const info = await eks.describeCluster({ name: `biomage-${config.clusterEnv}` }).promise();
+
+      const { name, endpoint } = info.cluster;
+      const certAuthority = info.cluster.certificateAuthority.data;
+      const roleArn = `arn:aws:iam::${accountId}:role/state-machine-role-${config.clusterEnv}`;
+
+      this.clusterInfo = {
+        accountId, name, endpoint, certAuthority, roleArn,
+      };
+    }
+
+    return this.clusterInfo;
+  }
+
+  static async getPipelineImage() {
+    const response = await fetch(
+      config.pipelineInstanceConfigUrl,
+      {
+        method: 'GET',
+      },
+    );
+
+    const txt = await response.text();
+    const manifest = YAML.parseAllDocuments(txt);
+
+    return jq.json(manifest, '..|objects|.r//empty');
+  }
+
+  async generatePipeline(experimentId) {
+    const { name, endpoint, certAuthority } = await this.getClusterInformation();
+    const namespace = `worker-${config.sandboxId}`;
+    const image = await PipelinesService.getPipelineImage();
+
+    return {
```

Comment:
I wonder if this is the best place to be keeping this job config?

Reply:
No, this will need to be refactored for local development anyways (or when a custom pipeline is run), so I thought I'd keep it here for now and find a new place for it in the next ticket.

Reply:
ok, if that is the plan, sounds good